### PR TITLE
error handler setting

### PR DESCRIPTION
**Pull request by denarch22**

### CODE REVIEW

**Comment by denarch22** [comment]
I am sending a review request for validation. I had problems with the PR because I had to resolve conflicts with some files. I have problems validating the periods because it tells me Error: Timeout was reached.

File: app/main.py [hunk 1]
```diff
@@ -26,14 +28,11 @@
...
+app.middleware("http")(log_requests)
 
+app.add_exception_handler(Exception, generic_exception_handler)
 app.add_exception_handler(
-    StarletteHTTPException, exception_handlers.http_exception_handler
-)
-app.add_exception_handler(
-    fastapi.exceptions.RequestValidationError,
-    exception_handlers.validation_exception_handler,
+    fastapi.exceptions.RequestValidationError, validation_exception_handler
```
Comment (erikasv) [inline app/main.py]: this line show a type checking error (vscode uses a [pydantic plugin](https://docs.pydantic.dev/latest/integrations/visual_studio_code/))

File: app/middleware/exception_handlers.py [hunk 2]
```diff
@@ -1,28 +1,53 @@
...
 
-async def http_exception_handler(request, exc):
+async def validation_exception_handler(
+    request: fastapi.Request, exc: fastapi.exceptions.RequestValidationError
+):
+    error_details = exc.errors()
+
+    first_error = error_details[0]
+    loc = first_error.get("loc", [])
+    expected = first_error.get("type", "unknown")
```
Comment (erikasv) [inline app/middleware/exception_handlers.py]: In the validation errors the attribute `type` doesn't seem to fit the "expected" descriptor, it usually refers to the type of error, for example: in `GET /metrics/LossPersistence/values?area_type=ea&area` the validation error is that "area_id is required", but type is `missing`, another example: in `POST /metrics/LossPersistence/values` if the bbox is missing, type is `value_error` which doesn't say anything about the expected input.

I suggest to remove the whole "expected" part

File: app/routes/metrics.py [hunk 3]
```diff
@@ -1,13 +1,14 @@
...
-
 from app.routes.schemas.polygon import Polygon
 from app.routes.schemas.MetricValues import MetricResponse
 import app.services.metrics as metrics_service
 from logging import getLogger
 from app.utils import context_vars
 
+
 request_id_context = context_vars.request_id_context
+logger = getLogger(__name__)
```
Comment (erikasv) [inline app/routes/metrics.py]: this variable is not used

File: app/routes/metrics.py [hunk 4]
```diff
@@ -26,7 +27,8 @@
     tags=["metrics"],
     responses={
         404: {"description": "Not found"},
-        422: {
+        500: {
+            "description": "Internal server error",
```
Comment (erikasv) [inline app/routes/metrics.py:29]: why did you remove the 422 error?

File: app/routes/metrics.py [hunk 5]
```diff
@@ -26,7 +27,8 @@
...
             "content": {
                 "application/json": {"example": validation_error_example}
```
Comment (erikasv) [inline app/routes/metrics.py]: this example is outdated

File: app/routes/schemas/polygon.py
Comment (erikasv) [inline app/routes/schemas/polygon.py:1]: validation for bbox presence (line 35) should be handled as the other validations

File: app/routes/schemas/polygon.py [hunk 6]
```diff
@@ -36,26 +38,20 @@ def __init__(self, **args):
     @field_validator("bbox", mode="before")
     def validate_bbox(cls, v):
         if len(v) not in [4, 6]:
-            raise ValueError("Bounding box (bbox) must have 4 or 6 elements.")
+            raise_http_exception(422, "bbox_length", url="None")
```
Comment (erikasv) [inline app/routes/schemas/polygon.py]: not sending the url is to lose information. If you let the error raise up to the exception_handlers it will not be lost

File: app/utils/errors.py [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+    method = "POST"
+
+    logger.error(
+        f"Error: {error_detail_template} - Path: {url} - Method: {method}",
+        extra={"request_id": request_id_context.get()},
+    )
+
+    raise fastapi.HTTPException(
+        status_code=status_code, detail={"message": error_detail}
+    )
```
Comment (erikasv) [inline app/utils/errors.py]: this function is per se an exception handler. It should be in the exception handlers file.

Instead of calling this function directly raise another type of error (if necessary) and handle it with this function.

**Review by erikasv — DISMISSED** [review]

**Comment by denarch22** [comment]
I reviewed the code and clarified questions with Erika, so I see the modification made well.

**Review by ManuelStardust — APPROVED** [review]
Everything works fine

**Comment by cazapatamar** [comment]
Late but everything works fine for me.
